### DISCUSSION TITLE
Collabora unable to verify certificate

### DISCUSSION THREAD

**Discussion by JulienPreuilh** [nextcloud/all-in-one#4296] [Questions] [answered]
Hi 

I completed installing NextCloud AIO being nginx proxy manager. I am able to get it to work, Windows and Android apps successfully connect to it. However it seems the collabora container can't.

I have seen several similar issues reported with NextCloud, but none under NextCloud AIO and I can't really understand the suggested fixes in those conversations to be honest. 

What would be different from standard setup is that I am not using a Let's Encrypt SSL that would be generated from Nginx Proxy Manager, I am using the SSL certificate generated by my website host. It is a wildcard let's encrypt certificate on my domain name, nginx proxy manager assigns NextCloud to a CNAME of that domain.
I see in the following pull request that this might be an issue : https://github.com/nextcloud/all-in-one/pull/4285
However, I am not sure how this translate to nginx proxy manager.

Below are screenshots of my npm config: 
![image](https://github.com/nextcloud/all-in-one/assets/75659112/f33343f2-89ad-4bb6-853b-f0c5d8c61696)
![image](https://github.com/nextcloud/all-in-one/assets/75659112/c6f05e4d-6b9e-437f-9fb4-94d381b2ec73)

How can I debug what the collabora container is doing ? Could it be unable to verify the wildcard certificate ?

**Comment by szaimen** [comment, marked as answer]
Hi, why do you think that collabora is not able to verify the cert? Is it not possible for you to use collabora inside Nextcloud? If so, can you follow https://github.com/nextcloud/all-in-one/discussions/1358?

**Reply by JulienPreuilh** [reply]
I was thinking collabora couldn't verify the certificate because I get the following error in the collabora container : 

```
wsd-00008-00015 2024-03-03 12:18:32.242531 +1100 [ remotefontconfig_poll ] ERR  Failed to lookup host [XXX]. Skipping (0: Success)| net/NetUtil.cpp:162
wsd-00008-00015 2024-03-03 12:18:32.242590 +1100 [ remotefontconfig_poll ] ERR  #-1: Failed to connect to XXX:443| net/HttpRequest.hpp:1282
wsd-00008-00015 2024-03-03 12:18:32.242605 +1100 [ remotefontconfig_poll ] ERR  Remote config server has response status code: 0 (Unknown)| wsd/COOLWSD.cpp:1383
```

and the following result from within the container : 

```
2ebe030610e8:/var/www/html# curl -vvv https://$NC_DOMAIN:443/hosting/discovery
* Host XXX:443 was resolved.
* IPv6: (none)
* IPv4: myIP
*   Trying myIP:443...
* Connected to XXX (myIP) port 443
* ALPN: curl offers h2,http/1.1
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
*  CAfile: /etc/ssl/certs/ca-certificates.crt
*  CApath: /etc/ssl/certs
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.3 (OUT), TLS alert, unknown CA (560):
* SSL certificate problem: unable to get local issuer certificate
* Closing connection
curl: (60) SSL certificate problem: unable to get local issuer certificate
More details here: https://curl.se/docs/sslcerts.html
```
I am unable to follow the instructions you mentioned, this is my NextCloud Office page : 
![image](https://github.com/nextcloud/all-in-one/assets/75659112/76423ace-360f-4820-827d-39d74711cbba)


I ticked _Disable certificate validation_, I was able to get to the Allow list for WOPI requests menu, but adding _,0.0.0.0/0_ didn't make the error disappear when I unticked that option.
